### DISCUSSION TITLE
MiNiksa is off to a new position

### DISCUSSION THREAD

**Discussion by miniksa** [microsoft/terminal#12801] [Announcements]
Folks, today I want to let you all know, my cherished GitHub community, that I'm heading off from the Terminal team to join a new role in a different organization at Microsoft. 

This wasn't an easy choice for me, having been embedded in this project from the beginning, but it should ultimately be the best move for me at this point in my career. It was my dream to get this code and this project onto GitHub where it could be public and open forever. And we're there. We've been successful. And it's not just us, it's all of you. Thank you for turning us into one of the most starred and followed projects published by Microsoft in open source. Thank you for contributing issues, discussions, and pull requests. Thank you for your patience and understanding with us as we continuously fought to balance that careful line between business necessity and customer willpower. 

The biggest thing I will miss about my new position is that it will be internal. I will no longer be performing on the world stage in total view and transparency of the developer community. I have to admit how strong a motivation that has been to always *Do The Right Thing* and how wonderful it feels to watch you all beam with joy when we release something beautiful into your hands. (And also how much I internalize and strive to do better every time you call us out on something we messed up on!)

However, you can bet that I will be taking everything I learned from my experience here in open source over the last few years and from my journey on this product to my new organization. This is only the end for here and for now. It will be my mission to get back to open source again in the future.

I know that I leave from a place of an immense crystalline knowledge about this system. While I will be unassigning myself from all work items and pull requests, transitioning them to my lovely team, you can freely call on me by @-ing my name going forward when you really need to know something that was in my head. I will be leaving that time and space for myself in the new role, especially as I'm in the immediate transition phase.

Reproduced below with limited redaction is the mail I sent to my internal teams and partners announcing this transition. In the true spirit of transparency, I share it with you as my last major posting on the way out.

Again, thanks to every single one of you. We shall meet again.

--Michael

<hr />

Friends, colleagues, partners, mentors:

Today I announce the end of an era in my life. An era that each of you were a part of in your own way. This message is as much about me as it is all of you, but mostly it's a message of appreciation. 

Back in 2014, I had the opportunity to take on a crusty and unloved part of Microsoft in the form of the Windows Console Host. I made quick changes without realizing the scope and often learned the hard way how crucial and impactful the console and its subsystem is to every piece of our line of business. Over the years, I met and worked with all of you on developing that derelict black box of text into the platform, application, and team that it is today in the Windows Terminal. Sometimes we worked together well; sometimes it was contentious. However, I owe all of you a debt of gratitude toward shaping the product and my career into what it is today.

To the console fans who drove the initial selfhosting, features, and spirit of revitalizing the console, thank you for being the initial guinea pigs, the loudest critics, and the biggest fans.
To the Windows Subsystems for Linux team, thank you for creating one of the coolest developer technologies ever, initiating the virtual terminal work in the console, and stealing the show at Build 2017.
To the VMs and containers teams who launched Docker and the gamut of technologies on Windows, thank you for stretching the boundaries of the console in runtime and VT and giving us the initial fodder to build ConPTY.
To the PowerShell and SSH teams, thank you for continuously challenging every aspect of the console on Windows from the APIs to eventually the VT sequences and PTY too.
To the OneCore and Recovery Console teams, thank you for all the work we did together in uniting the multiple copies of the console system into a unified, maintainable whole. Windows has one console to rule them all.
To the accessibility teams internal and external spanning Narrator, JAWS, and NVDA, thank you for all the work we’ve done on pushing the boundaries of MSAA and UIA to present the console and Terminal as beautifully with a screen reader as it is on a monitor.
To the performance teams, thank you for your unmatched set of analysis tooling and that constant voice you all left in the back of my head to make everything smaller, faster, lighter.
To the security teams including the Defender family of products, thank you for challenging us to invent performant ways of raising the bar for security in the wild-west of the “run anything you want” command-line environment.
To the compatibility teams where I started my career, thank you for instilling in me the challenge and passion for maintaining an operating system where what was written yesterday continues to work today and tomorrow. When you do it right, no one knows what you did...but I do.
To the application platform teams who continuously extended and built out the model for applications on Windows to increase the power and flexibility available to developer power customers, thank you.
To the input and windowing teams who rewrote our message loop half a dozen times, rewrote our IME connections half a dozen times, rewrote their own DPI change handlers half a dozen times, and then worked with us to rewrite them a half dozen more on both our side and in the OS side, thank you.
To the rendering teams providing a crash course in DirectX, Direct2D, and DirectWrite and fielding outlandish questions about fonts and drawing, thank you for teaching us and supporting us in our journey to learn more than we ever wanted to know about placing text on a screen.
To the globalization teams who taught the ins and outs of Unicode, codepages, and all the nitty gritty about text, thank you for instilling a constant thought about how customers outside the United States and the English language will experience the console and terminal.
To the data teams who helped us analyze and sort through the firehose of information we receive from our billions to trillions of daily activations, thank you for guiding our development and tracking our successes and failures. We couldn’t possibly focus without you.
To the XAML teams, thank you for working so tightly with us on expanding our premier UI platform to support developer needs and again for sharing and accelerating our engineering systems together.
To the Visual Studio and Visual Studio Code teams, thank you for joining forces with us on the Terminal control and the ConPTY. We’re all powered through the same console platform now and that’s a beautiful thing.
To the CRT and STL teams, thank you for always working so closely with us to resolve strange and interesting interactions between the client apps we power and the platforms we build.
To the GitHub community who made me truly believe in the power of open source and continuously awe me in new ways every day, thank you for your contributions, your generosity, your understanding, and your moments of teaching me who are customers truly are.
To the documentation teams, thank you for bringing us in early to open and public documentation, having our back on editing, cross-referencing, and flow, and working with us on our creative ideas for publicly publishing developer content.
To the community engagement and marketing teams, thank you for always having our back in the seas of the Internet and always pushing us to do better at building a community that wins developer mindshare for Microsoft.
To the platform architects and gurus, thank you for bearing with us as we awkwardly describe the strange and complex systems of the console and providing us your insights that saved us so much trouble.
To those of you who pushed and pushed your favorite bug or feature in the console and/or terminal system, thank you for being relentless advocates for a better product for yourselves and those developers you represent.
To the early in career and Squad Finance crew, thank you for giving me a platform to share one of my favorite topics, personal finance, with others and teach those new to the company everything I wish I knew about money 12 years ago when I started here.
To all the rest of the "Win the Developer Box" crew that I haven’t already mentioned one way or another, thank you for making work a true joy and pleasure every day. We were truly a family.
To Dustin Howett (@DHowett), team lead in transforming the system in the Terminal today, thank you for your true excitement from the first day you joined the team and for every day you and I remained on the same wavelength as we delivered the ultimate terminal experience for Windows. I’m glad to call us not only lead and report, but also partners and friends.
And to anyone I may have missed, by now you know the game: Thank you.

Eight years of my heart and soul went into this product and my relationships with all of you. And all of you went into it as well. I think it shows. These products wouldn’t have been anywhere without my work, architecture, and guidance as we changed teams, managers, and priorities over the years. But they also wouldn’t have been here without all of you working with me and my team.

If I can leave you with anything, it's that the people are truly what make the difference. When people can trust each other, band together, and deliver, we engender trust in our customers and we win. Our products may be what we sell, but they're all a reflection of the people who build them. The most attractive product is the one made with genuine care. Our customers, the developer community, can feel that in their core as they choose us. Our fellow engineers can feel that when they choose to work with us. Focus on the people and everything else will come together.

Thank you for reflecting with me today as I move on to a new era in the Engineering Systems team. A new challenge I am thrilled to join and strive to establish as strong a legacy there as the Console and Terminal have brought to all of us.

My last day on the Terminal team will be 2022-04-08. Then I will experience "no freer time than the time between jobs" with a 3 week “April 2020” family vacation followed by a start on my new team 2022-05-02.

I’ve left you all in the care of the other amazing engineers on the Terminal team. You can count on them to continue the same spirit, drive, and passion for this product, the people they work with, and the developer community. I mentored all of them. :P

--Michael

**Comment by llama-for3ver** [comment]
❤

**Comment by bitcrazed** [comment]
Sincerest thanks, @miniksa for all your hard work over the last 8 years, and for your patience, honesty, candor, persistence, sheer bloody-minded determination, and your "screw it - I'm going to do the right thing - you guys fix the politics" attitude without which we wouldn't have been able to overhaul the Windows Console and command-line infrastructure, [build ConPTY](https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/), or [introduce a new, modern Windows Terminal app](https://www.youtube.com/watch?v=KMudkRcwjCw).

Working with you and the Terminal team has been a major highlight of my career that I will never forget and always treasure.

Wishing you every success in your new role - I have no doubt that you'll continue to be awesome and bring your unbridled clarity and amazing skills to bear on whatever challenges come your way. 

Stay in touch!

**Comment by ghost** [comment]
I discovered this project late, i didnt know who you were and I'm sorry for that, and whatever you do, i wish you the best luck.

Thanks for this amazing project!
